### PR TITLE
Auth efcore eigth

### PR DESCRIPTION
**Pull request by sahma19**

### CODE REVIEW

File: domains/auth/Auth.API/API/Program.cs [hunk 1]
```diff
@@ -174,7 +173,7 @@
 }
 catch (Exception e)
 {
-    app.Logger.LogError(e, "An exception has occurred while starting up.");
+    app.Logger.LogError(e, "An exception has occurred while starting up");
```
Comment (tnickelsen) [inline domains/auth/Auth.API/API/Program.cs:176]: Why this change?

File: domains/auth/Auth.API/Integration.Tests/AuthWebApplicationFactory.cs [hunk 2]
```diff
@@ -43,16 +43,16 @@ protected override void ConfigureWebHost(IWebHostBuilder builder)
         {
             services.Remove(services.First(x => x.ServiceType == typeof(DbContextOptions<DataContext>)));
             services.Remove(services.First(x => x.ServiceType == typeof(DataContext)));
-            services.Remove(services.First(x => x.ServiceType == typeof(NpgsqlDataSourceBuilder)));
-            services.AddSingleton(new NpgsqlDataSourceBuilder(testContainer.GetConnectionString()));
-            services.AddDbContext<DataContext>((serviceProvider, options) => options.UseNpgsql(serviceProvider.GetRequiredService<NpgsqlDataSourceBuilder>().Build()));
+            services.AddDbContext<DataContext>(options => options
+                .UseNpgsql(DataContext.GenerateNpgsqlDataSource(testContainer.GetConnectionString()))
+                .ConfigureWarnings(warnings => warnings.Ignore(CoreEventId.ManyServiceProvidersCreatedWarning)));
```
Comment (tnickelsen) [inline domains/auth/Auth.API/Integration.Tests/AuthWebApplicationFactory.cs:48]: Why ignore the warnings here?

File: domains/auth/Auth.API/Integration.Tests/RollbackMigrationTests.cs [hunk 3]
```diff
@@ -12,10 +10,9 @@ public class RollbackMigrationTests : IClassFixture<AuthWebApplicationFactory>
     public RollbackMigrationTests(AuthWebApplicationFactory factory) => this.factory = factory;
 
     [Fact]
-    public void Migrate_ShouldRollbackMigrationsAndThenApplyMigrations_WhenInvokedWithZeroAndThenNoArguments()
+    public async void Migrate_ShouldRollbackMigrationsAndThenApplyMigrations_WhenInvokedWithZeroAndThenNoArguments()
     {
-        var migrator = factory.DataContext.Database.GetService<IMigrator>(); //Factory migrates the database to latest migration automatically
-        migrator.Migrate("0"); //This migrates down to migration no. 0 aka. remove all migrations
-        migrator.Migrate(); //This migrates the database to latest migration (again)
```
Comment (tnickelsen) [inline domains/auth/Auth.API/Integration.Tests/RollbackMigrationTests.cs:19]: Keep the migrator code. Good with MigrateAsync()

File: domains/auth/Auth.API/API/Program.cs (hunk 1 above)
Comment (sahma19) [reply domains/auth/Auth.API/API/Program.cs:176]: To remove a warning. Compiler Hates ".", in error messages

File: domains/auth/Auth.API/Integration.Tests/AuthWebApplicationFactory.cs (hunk 2 above)
Comment (sahma19) [reply domains/auth/Auth.API/Integration.Tests/AuthWebApplicationFactory.cs:48]: The warnings it produces is a product of the upgrade to ef-core 8. Microsoft themselves suggests, that it can be ignored. AKA, it seems to be a false positive.

**Review by tnickelsen — APPROVED** [review]
